CONTRIBUTING.md

```diff
@@ -1,50 +1,161 @@
+cd ODE-Test-Problems/
+```
+
+2. Create a new folder in the `src/+opt/` directory. Follow the same naming conventions as the existing problems. Start the name with `+` to maintain the structure of the Matlab/Octave package. Also create a subfolder named `+presets` in the new problem folder:
+
+```bash
+mkdir src/+opt/+newtest
+mkdir src/+opt/+newtest/+presets
+cd src/+opt/+newtest/
+```
+
+3. The minimal set of files needed inside the problem folder to set up a new test problem are:
+  *  The right-hand-side function named as `f.m` 
+  *  The problem class to initialize problem objects and its methods and properties
+  *  The parameters class defines the parameters of the new problem
+  *  A `Canonical.m` preset inside the `+presets` subfolder to set the initial condition and parameters in your case
+
+## The right-hand-side structure
 
-### Problem Class
+The right-hand-side structure providesvarious derivatives of the problem. They are implemented in separate function files. The right-hand-side function `f.m`, which is the time-derivative of the state `y` is defined as a function with at least two arguments `f(t,y)`. If the right-hand-side function needs other parameters they can also be passed to this function:
+
+```Matlab
+
+function dy = f(t, y, param1, ...)
+  dy  = ... 
+end
+
+```
+
+For more information about this formulation please refer to our [paper](https://github.com/ComputationalScienceLaboratory/ODE-Test-Problems/blob/master/paper/paper.md).
+
+## The problem class 
 
 A problem package must contain a class named `<Name>Problem.m` that is a subclass of `otp.Problem`. There are two
-methods that must be implemented: the constructor and `onSettingsChanged`. Optionally, one can override functions such
-as `internalPlot` and `internalSolve` to provide problem-specific defaults. Partitioned problems can add custom `RHS`'s
+methods that must be implemented: the constructor and `onSettingsChanged`. Optionally, one can override functions such as `internalPlot` and `internalSolve` to provide problem-specific defaults. Partitioned problems can add custom right-hand-side functions
 as class properties with private write access. The property name should start with `RHS`, e.g., `RHSStiff`.
 
-### Parameters
+The template for a new class of problems called `SimpleProblem` looks like:
 
-A problem package must also contain a class named `<Name>Parameters.m`. It only needs to provide public properties for
-each of the problem parameters; no constructor or methods are needed. Property validation is currently not supported in
-Octave. Therefore, we use a custom comment syntax that is parsed by the installer to optionally include validation.
+```Matlab
 
-```matlab
-% Example
-properties
-    MyProp %MATLAB ONLY: (1,1) {mustBeInteger, mustBePositive}
+classdef SimpleProblem < otp.Problem
+
+    methods
+        function obj = SimpleProblem(timeSpan, y0, parameters)
+            obj@otp.Problem('Simple Test Problem', [], timeSpan, y0, parameters);
```

Comment:
('New Test', []

I think we agreed to not have 'Problem' In this internal name.